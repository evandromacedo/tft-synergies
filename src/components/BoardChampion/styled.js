import styled from 'styled-components';
import TierBorder from '../Common/TierBorder';

export const Wrapper = styled(TierBorder)`
  position: relative;
  height: 12.8rem;
  padding: 0.8rem;
  max-width: 25.6rem;
  display: grid;
  grid-gap: 0.8rem;

  /* Background image for Splash Art */
  &:before {
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    z-index: -1;
    background-image: url(${props => props.background});
    background-position: center right;
    background-size: cover;
  }
`;

export const Synergies = styled.div`
  display: grid;
  grid-gap: 0.8rem;
  grid-template-rows: repeat(3, 2rem);
`;

export const Items = styled.div`
  display: grid;
  grid-gap: 0.8rem;
  grid-template-columns: 2.8rem 2.8rem 2.8rem 1fr;
  align-items: center;
`;

export const Item = styled.div`
  width: 2.8rem;
  height: 2.8rem;
  background-color: ${props => props.theme.background3};
`;

export const Name = styled.p`
  ${props => props.theme.bold20px}
  text-align: center;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
`;
