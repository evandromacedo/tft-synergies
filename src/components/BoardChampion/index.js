import React from 'react';
import PropTypes from 'prop-types';
import * as S from './styled';
import ClassOrOrigin from '../ClassOrOrigin';

// This will be made again to atempt the API and props.
// Must implement the drag and drop functionality afterwards.
function BoardChampion({ champion, background }) {
  return (
    <S.Wrapper cost={champion.cost} background={background}>
      <S.Synergies>
        {champion.synergies.map((synergy, index) => (
          <ClassOrOrigin key={index} type={synergy} />
        ))}
      </S.Synergies>
      <S.Items>
        <S.Item />
        <S.Item />
        <S.Item />
        <S.Name>{champion.name}</S.Name>
      </S.Items>
    </S.Wrapper>
  );
}

BoardChampion.propTypes = {
  champion: PropTypes.shape({
    name: PropTypes.string,
    cost: PropTypes.number,
    synergies: PropTypes.arrayOf(PropTypes.string)
  }).isRequired,
  background: PropTypes.string
};

export default BoardChampion;
