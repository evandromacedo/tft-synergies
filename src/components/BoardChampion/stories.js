import React from 'react';
import BoardChampion from './index.js';
import { storiesOf } from '@storybook/react';

storiesOf('BoardChampion', module)
  .add('Kassadin', () => (
    <BoardChampion
      champion={{ name: 'Kassadin', cost: 1, synergies: ['void', 'sorcerer'] }}
      background={
        'http://raw.communitydragon.org/latest/game/assets/ux/tft/championsplashes/tft_kassadin.png'
      }
    />
  ))
  .add('Ahri', () => (
    <BoardChampion
      champion={{ name: 'Ahri', cost: 2, synergies: ['wild', 'sorcerer'] }}
      background={
        'http://raw.communitydragon.org/latest/game/assets/ux/tft/championsplashes/tft_ahri.png'
      }
    />
  ))
  .add('Aatrox', () => (
    <BoardChampion
      champion={{ name: 'Aatrox', cost: 3, synergies: ['demon', 'blademaster'] }}
      background={
        'http://raw.communitydragon.org/latest/game/assets/ux/tft/championsplashes/tft_ahri.png'
      }
    />
  ))
  .add('Akali', () => (
    <BoardChampion
      champion={{ name: 'Akali', cost: 4, synergies: ['ninja', 'assassin'] }}
      background={
        'http://raw.communitydragon.org/latest/game/assets/ux/tft/championsplashes/tft_akali.png'
      }
    />
  ))
  .add('Anivia', () => (
    <BoardChampion
      champion={{ name: 'Anivia', cost: 5, synergies: ['glacial', 'elementalist'] }}
      background={
        'http://raw.communitydragon.org/latest/game/assets/ux/tft/championsplashes/tft_anivia.png'
      }
    />
  ));
