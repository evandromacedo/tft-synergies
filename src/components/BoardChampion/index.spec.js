import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';
import BoardChampion from '.';

const mockKassadin = { name: 'Kassadin', cost: 1, synergies: ['void', 'sorcerer'] };

describe('<BoardChampion />', () => {
  it('render properly and matches snapshot', () => {
    const wrapper = shallow(
      <BoardChampion
        champion={mockKassadin}
        background={
          'http://raw.communitydragon.org/latest/game/assets/ux/tft/championsplashes/tft_kassadin.png'
        }
      />
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
