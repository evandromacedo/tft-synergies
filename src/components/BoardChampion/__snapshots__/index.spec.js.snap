// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoardChampion /> render properly and matches snapshot 1`] = `
<Styled(styled.div)
  background="http://raw.communitydragon.org/latest/game/assets/ux/tft/championsplashes/tft_kassadin.png"
  cost={1}
>
  <styled.div>
    <ClassOrOrigin
      key="0"
      type="void"
    />
    <ClassOrOrigin
      key="1"
      type="sorcerer"
    />
  </styled.div>
  <styled.div>
    <styled.div />
    <styled.div />
    <styled.div />
    <styled.p>
      Kassadin
    </styled.p>
  </styled.div>
</Styled(styled.div)>
`;
